File: leetcode/weekly-contests/contest-177/1360-number-of-days-between-two-dates/sol.cpp
class Solution {
public:
    int days[12] = {31 , 28 , 31 , 30 , 31 , 30 , 31 , 31 , 30 , 31 , 30 , 31} ;
    int daysBetweenDates(string date1, string date2) ;
    int dayFrom1971(string date) ;
    bool isLeap(int year) ;
};

bool Solution :: isLeap(int year)
{
    return (year%4 == 0 and (year%100 != 0 || year%400 == 0)) ;
}
int Solution :: dayFrom1971(string date)
{
    int year = stoi(date.substr(0 , 4)) ;
    int month = stoi(date.substr(5 , 2)) ;
    int day = stoi(date.substr(8)) ;
    
    for(int y = 1971 ; y < year ; y++)
    {
        day += (isLeap(y) ? 366 : 365 );
    }
    
    if(isLeap(year) and month > 2)
    {
        day ++ ;
    }
    day += accumulate(begin(days) , begin(days) + month - 1 , 0) ;
    return day ;
}

int Solution :: daysBetweenDates(string date1 , string date2)
{
    return abs(dayFrom1971(date1) - dayFrom1971(date2)) ;
}

File: leetcode/weekly-contests/contest-177/1360-number-of-days-between-two-dates/sol.py
class Solution:
    days = [31 , 28 , 31 , 30 , 31 , 30 , 31 , 31 , 30 , 31 , 30 , 31]
    def daysFrom1971(self , date : str) -> int :
        year = int(date[:4])
        month = int(date[5:7])
        day = int(date[8:])
        for y in range(1971 , year) :
            if (y%4 == 0 and (y%100 != 0 or y%400 == 0)) :
                day += 366
            else :
                day += 365
        
        if month > 2 and year%4 == 0 and (year%100 != 0 or year%400 == 0) :
            day += 1
        
        while month > 1 :
            month -= 1
            day += self.days[month - 1]
        return day
    def daysBetweenDates(self, date1: str, date2: str) -> int :
        return abs(self.daysFrom1971(date1) - self.daysFrom1971(date2))
        
        
        
